Implementation/FindMyBLEDevice/FindMyBLEDevice/Services/Bluetooth/RssiToMeter.cs

```diff
@@ -0,0 +1,114 @@
+﻿// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2022 Leo Köberlein <leo@wolfgang-koeberlein.de>
+
+using FindMyBLEDevice.Models;
+using FindMyBLEDevice.Services.Settings;
+using FindMyBLEDevice.Services.Geolocation;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace FindMyBLEDevice.Services.Bluetooth
+{
+    public class RssiBuffer
+    {
+        protected double rssi = 0;
+
+        public virtual double Get()
+        {
+            return this.rssi;
+        }
+
+        public virtual void Update(double newRssi)
+        {
+            this.rssi = newRssi;
+        }
+    }
+
+    public class RssiRingBuffer : RssiBuffer 
+    {
+        protected readonly ISettings settings;
+        protected readonly List<double> rssiBuff;
+
+        public RssiRingBuffer(ISettings settings)
+        {
+            this.settings = settings;
+            rssiBuff = new List<double>();
+        }
+
+        public override void Update(double newRssi)
+        {
+            rssiBuff.Add(newRssi);
+            int rssiInterval = settings.Get(SettingsNames.RssiInterval, Constants.RssiIntervalDefault);
+            int buffSize = rssiInterval > 0
+                ? Math.Min(Constants.RssiBufferDuration / rssiInterval, Constants.RssiBufferMaxSize)
+                : Constants.RssiBufferMaxSize;
+            while (rssiBuff.Count > buffSize) rssiBuff.RemoveAt(0);
+            base.Update(rssiBuff.Average());
+        }
+    }
+
+    public class RssiExponentialBuffer : RssiBuffer
+    {
+        // meters per degree of lat/lon = 2 * PI * radius of earth / 360 degrees
+        protected const double metersPerDegree = 2 * Math.PI * 6371000 / 360;
+        // history-factor for exponential moving average
+        protected const double base_alpha = 0.95;
+        // how much of the non-history factor (1 - alpha) is dependent on GPS movement
+        protected const double beta = 0.75;
+        protected const double base_movement = 10;
```

Comment:
Maybe some more explanation on why you decided for value `10` here would be nice.